addons/gamification/models/gamification_goal.py

```diff
@@ -102,17 +102,19 @@ def _check_remind_delay(self):
 
     def _get_write_values(self, new_value):
         """Generate values to write after recomputation of a goal score"""
-        if new_value == self.current:
```

Comment:
We should probably still return `{}` if `result` is empty after the next steps.

Reply:
Ok, done. I also updated the `update_goal` method to use the same format of `goals_to_write.update` as the other calls.